### ISSUE TITLE
FileSystem and objectStore are being created successfully only on first cluster in a multiple rook cluster environment

### ISSUE THREAD

**Issue by dangula** [rook/rook#970]
When deploying multiple rook cluster (different cluster name and namespace), FileSystem and objectStore are being successfully created in the first cluster that was deployed, and failing in other clusters. 

**Repro Steps**
1. Install Rook Operator
2. Install 2 Rook clusters  - with different cluster names and in different name spaces.
3. Create FileSystem and ObjectStore for cluster 1
4. Create FileSystem and ObjectStore for cluster 2

**Expected Result**
 FileSystem and object Store is created successfully in both clusters. 

**Actual Result**
FileSystem and ObjectStore not created for cluster 2, errors in rook-api service. 

_Additional Data_ 
Snippit from Api service pod logs from failed cluster
```
 exec: Running command: ceph osd pool application enable test-fs-2-metadata cephfs --cluster=rook-cluster-2 --conf=/var/lib/rook/rook-cluster-2/rook-cluster-2.config --keyring=/var/lib/rook/rook-cluster-2/client.admin.keyring --format json --out-file /tmp/883951700
2017-09-07 21:12:27.739241 I | exec: enabled application 'cephfs' on pool 'test-fs-2-metadata'
2017-09-07 21:12:27.739412 I | cephclient: creating pool test-fs-2-metadata succeeded, buf: 
2017-09-07 21:12:27.739562 I | exec: Running command: ceph fs new test-fs-2 test-fs-2-metadata test-fs-2-data --cluster=rook-cluster-2 --conf=/var/lib/rook/rook-cluster-2/rook-cluster-2.config --keyring=/var/lib/rook/rook-cluster-2/client.admin.keyring --format json --out-file /tmp/080018851
2017-09-07 21:12:28.335082 I | exec: new fs with metadata pool 3 and data pool 2
2017-09-07 21:12:28.335403 I | cephmds: created file system test-fs-2 on data pool test-fs-2-data and metadata pool test-fs-2-metadata
2017-09-07 21:12:28.335436 I | api-k8s: Starting the MDS
2017-09-07 21:12:28.335442 I | op-mds: start running mds
2017-09-07 21:12:28.348522 E | api: failed to start mds: failed to create mds keyring. failed to get mds secrets. User "system:serviceaccount:rook-cluster-2:rook-api" cannot get secrets in the namespace "rook-cluster-2". (get secrets rook-ceph-mds)
2017-09-07 21:12:28.348570 I | api: 69bd4d66-8a03-4660-a342-5302af11d9bb	POST	/filesystem	CreateFileSystem	500 Internal Server Error	5.066915744s
.
.
2017-09-07 21:13:10.567447 I | exec: Running command: ceph osd pool set default-c2.rgw.buckets.data size 1 --cluster=rook-cluster-2 --conf=/var/lib/rook/rook-cluster-2/rook-cluster-2.config --keyring=/var/lib/rook/rook-cluster-2/client.admin.keyring --format json --out-file /tmp/822476748
2017-09-07 21:13:11.602351 I | exec: set pool 9 size to 1
2017-09-07 21:13:11.602503 I | exec: Running command: ceph osd pool application enable default-c2.rgw.buckets.data default-c2.rgw.buckets.data --cluster=rook-cluster-2 --conf=/var/lib/rook/rook-cluster-2/rook-cluster-2.config --keyring=/var/lib/rook/rook-cluster-2/client.admin.keyring --format json --out-file /tmp/246867643
2017-09-07 21:13:12.614549 I | exec: enabled application 'default-c2.rgw.buckets.data' on pool 'default-c2.rgw.buckets.data'
2017-09-07 21:13:12.614659 I | cephclient: creating pool default-c2.rgw.buckets.data succeeded, buf: 
2017-09-07 21:13:12.616939 E | api: failed to create object store: failed to start rgw. failed to create rgw keyring. failed to get rgw secrets. User "system:serviceaccount:rook-cluster-2:rook-api" cannot get secrets in the namespace "rook-cluster-2". (get secrets rook-ceph-rgw-default-c2)
2017-09-07 21:13:12.616964 I | api: bf634e12-6eea-4876-beb0-be67eff511fc	POST	/objectstore	CreateObjectStore	500 Internal Server Error	18.740481423s


```
please look at the attached file for complete logs
[rook-api.txt](https://github.com/rook/rook/files/1286167/rook-api.txt)
[All logs.zip](https://github.com/rook/rook/files/1286170/All.logs.zip)

**Comment by travisn** [comment]
looks like an RBAC issue. Is this fixed now in #972? I thought i saw some rbac permissions created for other namespaces.

**Comment by dangula** [comment]
I am seeing service account created for both clusters, but clusterrolebinding created only for cluster 1. Service accounts, cluster roles are clusterrole bindings are created during cluster deploy time. 
```
$ **kubectl get serviceaccounts --all-namespaces**
NAMESPACE        NAME            SECRETS   AGE
default          default         1         43m
default          rook-operator   1         16m
kube-public      default         1         43m
kube-system      default         1         43m
test-cluster-1   default         1         15m
test-cluster-1   rook-api        1         15m
test-cluster-2   default         1         15m
test-cluster-2   rook-api        1         14m
```

```
$ **kubectl get secret --all-namespaces -o wide**
NAMESPACE        NAME                        TYPE                                  DATA      AGE
default          default-token-9xq0k         kubernetes.io/service-account-token   3         47m
default          rook-operator-token-p7qgt   kubernetes.io/service-account-token   3         20m
default          test-cluster-1-rook-user    kubernetes.io/rbd                     1         19m
default          test-cluster-2-rook-user    kubernetes.io/rbd                     1         18m
kube-public      default-token-0dwcj         kubernetes.io/service-account-token   3         47m
kube-system      default-token-1kbp8         kubernetes.io/service-account-token   3         47m
test-cluster-1   default-token-f6kww         kubernetes.io/service-account-token   3         19m
test-cluster-1   rook-admin                  kubernetes.io/rbd                     1         19m
test-cluster-1   rook-api-token-prvq0        kubernetes.io/service-account-token   3         19m
test-cluster-1   rook-ceph-mds               kubernetes.io/rook                    1         18m
test-cluster-1   rook-ceph-mgr0              kubernetes.io/rook                    1         19m
test-cluster-1   rook-ceph-mon               kubernetes.io/rook                    4         19m
test-cluster-2   default-token-xw0b9         kubernetes.io/service-account-token   3         19m
test-cluster-2   rook-admin                  kubernetes.io/rbd                     1         19m
test-cluster-2   rook-api-token-cwl8l        kubernetes.io/service-account-token   3         18m
test-cluster-2   rook-ceph-mgr0              kubernetes.io/rook                    1         18m
test-cluster-2   rook-ceph-mon               kubernetes.io/rook                    4         19m
```

```
**$kubectl get clusterrolebinding -o wide**                                                                                                                                                                                                                          

NAME                                           AGE       ROLE                                           USERS                            GROUPS                                         SERVICEACCOUNTS
anon-user-access                               20m       cluster-admin                                  system:anonymous
cluster-admin                                  48m       cluster-admin                                                                   system:masters
rook-api                                       20m       rook-api                                                                                                                       test-cluster-1/rook-api
rook-operator                                  20m       rook-operator                                                                                                                  default/rook-operator
system:basic-user                              48m       system:basic-user                                                               system:authenticated, system:unauthenticated
system:controller:attachdetach-controller      48m       system:controller:attachdetach-controller                                                                                      kube-system/attachdetach-controller
system:controller:certificate-controller       48m       system:controller:certificate-controller                                                                                       kube-system/certificate-controller
system:controller:cronjob-controller           48m       system:controller:cronjob-controller                                                                                           kube-system/cronjob-controller
system:controller:daemon-set-controller        48m       system:controller:daemon-set-controller                                                                                        kube-system/daemon-set-controller
system:controller:deployment-controller        48m       system:controller:deployment-controller                                                                                        kube-system/deployment-controller
system:controller:disruption-controller        48m       system:controller:disruption-controller                                                                                        kube-system/disruption-controller
system:controller:endpoint-controller          48m       system:controller:endpoint-controller                                                                                          kube-system/endpoint-controller
system:controller:generic-garbage-collector    48m       system:controller:generic-garbage-collector                                                                                    kube-system/generic-garbage-collector
system:controller:horizontal-pod-autoscaler    48m       system:controller:horizontal-pod-autoscaler                                                                                    kube-system/horizontal-pod-autoscaler
system:controller:job-controller               48m       system:controller:job-controller                                                                                               kube-system/job-controller
system:controller:namespace-controller         48m       system:controller:namespace-controller                                                                                         kube-system/namespace-controller
system:controller:node-controller              48m       system:controller:node-controller                                                                                              kube-system/node-controller
system:controller:persistent-volume-binder     48m       system:controller:persistent-volume-binder                                                                                     kube-system/persistent-volume-binder
system:controller:pod-garbage-collector        48m       system:controller:pod-garbage-collector                                                                                        kube-system/pod-garbage-collector
system:controller:replicaset-controller        48m       system:controller:replicaset-controller                                                                                        kube-system/replicaset-controller
system:controller:replication-controller       48m       system:controller:replication-controller                                                                                       kube-system/replication-controller
system:controller:resourcequota-controller     48m       system:controller:resourcequota-controller                                                                                     kube-system/resourcequota-controller
system:controller:route-controller             48m       system:controller:route-controller                                                                                             kube-system/route-controller
system:controller:service-account-controller   48m       system:controller:service-account-controller                                                                                   kube-system/service-account-controller
system:controller:service-controller           48m       system:controller:service-controller                                                                                           kube-system/service-controller
system:controller:statefulset-controller       48m       system:controller:statefulset-controller                                                                                       kube-system/statefulset-controller
system:controller:ttl-controller               48m       system:controller:ttl-controller                                                                                               kube-system/ttl-controller
system:discovery                               48m       system:discovery                                                                system:authenticated, system:unauthenticated
system:kube-controller-manager                 48m       system:kube-controller-manager                 system:kube-controller-manager
system:kube-dns                                48m       system:kube-dns                                                                                                                kube-system/kube-dns
system:kube-scheduler                          48m       system:kube-scheduler                          system:kube-scheduler
system:node                                    48m       system:node                                                                     system:nodes
system:node-proxier                            48m       system:node-proxier                            system:kube-proxy
```

**Comment by travisn** [comment]
@dangula does this still repro since #1029

**Comment by jbw976** [comment]
I just updated the integration tests in my PR branch #1077 and it would be really nice to have this test, which would install one filestore cluster and one bluestore cluster.  I'm going to try to enable it and see if it now passes.

**Comment by jbw976** [comment]
I was able to get multiple clusters working and the integration test for it, so I'm adding this to 0.6

**Comment by jbw976** [comment]
Fixed in #1077
